### DISCUSSION TITLE
PHP 8.1

### DISCUSSION THREAD

**Discussion by wocky68** [scheb/open-apexx#54] [General]
Hallo Zusammen,
es gibt nicht zufällig ein Update für PHP 8.1 ?
3 Seiten von mir sind down...... 
Ich versuche einfach mal die Anfrage bei euch

**Comment by bigfraggle** [comment]
Mal die Fehlermeldungen aktiviert und geschaut, wo das Problem konkret liegt?

**Comment by wocky68** [comment]
Nein, kannst du mir sagen wie ich das mache?Von meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: Marc ***@***.***> Datum: 25.11.22  17:52  (GMT+01:00) An: scheb/open-apexx ***@***.***> Cc: wocky68 ***@***.***>, Author ***@***.***> Betreff: Re: [scheb/open-apexx] PHP 8.1 (Issue #51) 
Mal die Fehlermeldungen aktiviert und geschaut, wo das Problem konkret liegt?

**Comment by bigfraggle** [comment]
War irgendwo in der lib/config.php.
Und dann noch in der php.ini "display_errors = On" setzen.

**Comment by wocky68** [comment]
Ich finde keine php.ini.....Das andere habe ich gefunden

**Comment by bigfraggle** [comment]
Dann leg die Datei mit einem Editor an, schreibe "display_errors = On" rein und pack sie ins Hauptverzeichnis.

**Comment by wocky68** [comment]
Habe ich gemacht aber kommt nichts...

**Comment by wocky68** [comment]
So sieht das in der Datei config.php aus

 

// DEBUG
////////////////////////////////////////////////////////////////////////////
///////

// Kritische Fehlermeldungen anzeigen (true/false)

$set['showerror'] = true;

// Fehler-Report am Ende zeigen (true/false)

$set['errorreport'] = true;

// ini_set('display_errors', 1);

// ini_set('display_startup_errors', 1);

// error_reporting(E_ALL);

// Cache immer ausgeben (true/false)

$set['outputcache'] = false;

// Renderzeit anzeigen (true/false)

$set['rendertime'] = false;

// Anfang und Ende der Templates anzeigen

// 0 = aus

// 1 = durch HTML-Kommentare

// 2 = sichtbare Rahmen

$set['tmplwhois'] = 0;

ini_set("default_charset", "ISO-8859-1");

?>

**Comment by wocky68** [comment]
Noch eine Idee?Sehe leider keine Fehlermeldungen....

**Comment by bigfraggle** [comment]
Und mit PHP 8.0 lief es bei dir?

**Comment by wocky68** [comment]
Nein, habe auf PHP 8.0 umgestellt tat sich aber auch nichts....

**Comment by bigfraggle** [comment]
Naja. Hast du mal die Änderungen für PHP 8.0 installiert? Gibt in den geschlossenen Beiträgen dazu ja schon einiges...

**Comment by wocky68** [comment]
Ok.... leider bin ich darin gar nicht bewandert.... Ich kann etwas erledigen oder in Scripten etwas ändern usw...Aber was irgendwas bewirkt weiss ich leider nicht....Ich habe das CMS für 3 Vereins Homepage benutzt ohne Schnickschnack....Wäre total lieb wenn du mir hilfst...Vielleicht auch außerhalb von Github, ist nicht so kompliziert dann....Du könntest die Lösung dann hier Online stellen....

**Comment by bigfraggle** [comment]
Ich bin in Sachen PHP auch nicht so bewandert. Ich kann zwar Vieles nachvollziehen, aber selber machen dann doch weniger.

Wenn du es noch nicht gemacht hast, solltest du hier: https://github.com/scheb/open-apexx/tree/master/src ... alle Dateien runterladen, die im letzten Jahr aktualisiert wurden ("Fixes for PHP8") und bei dir austauschen. Dann läuft es zumindest unter PHP8. Ob es dann mit 8.1 weitere Probleme gibt, sieht man dann.

**Comment by wocky68** [comment]
Habe jetzt alle Template aus deinem genannten Pfad ausgetauscht und es kommt immer noch nichts....Alle Templates wo der Hinweis mit PHP dahinter stand....Habe beim Provider auf PHP 8.0 umgestellt 😭😭😭Noch eine Idee?www.tambourcorps-eller.de ist die Seite

**Comment by bigfraggle** [comment]
Bei einer ganz normalen Standardinstallation läuft Apexx bei mir mit den geänderten Dateien. Du solltest das mit der php.ini wie oben erwähnt umsetzen, damit die Fehler auch angezeigt werden.

Andererseits läuft mein Hauptprojekt mit PHP8 auch nicht rund - aber da habe ich auch diverses PHP innerhalb der Templates. Bin mir nicht sicher, ob ich Zeit und Muße habe, das alles zu überprüfen. Vielleicht lasse ich da auch einfach PHP 7.4 laufen - auch wenn sich das der Provider bezahlen lässt.

Grundsätzlich sollten wir alle überlegen, ob es auf Dauer noch Sinn macht, mit Apexx weiter zu arbeiten. Da @scheb sich nicht mehr kümmert und das Know-How unter den Nutzern begrenzt ist, wird irgendwann eh der Punkt kommen, an dem man auf ein anderes System umsteigen muss.

**Comment by bigfraggle** [comment]
Ansonsten schreib mal ne Mail an marc punkt friedrichs at outlook punkt com
